test/cypress/integration/group-proposals.spec.js

```diff
@@ -236,33 +236,54 @@ describe('Proposals - Add members', () => {
       })
   })
 
-  it('It should display that no one is doing any contribution', () => {
+  it('User click on contribution page and see the card', () => {
     cy.getByDT('contributionsLink').click()
     cy.getByDT('addIncomeDetailsFirstCard').should('contain', 'Add your income details')
+  })
+
+  it('Open the income detail modal should ask only if you want to receive or pledge', () => {
     cy.getByDT('openIncomeDetailModal').click()
     cy.getByDT('introIncomeOrPledge').should('not.be.visible')
+  })
+
+  it('After selecting the pledge option, it should ask you how much you want to pledge', () => {
     cy.getByDT('dontNeedsIncomeRadio').click()
     cy.getByDT('introIncomeOrPledge').should('contain', 'How much do you want to pledge?')
+  })
+
+  it('After selecting the amount and close the modal it should show that no one is in need', () => {
     cy.getByDT('inputIncomeOrPledge').type(500)
     cy.getByDT('submitIncome').click()
     cy.getByDT('receivingParagraph').should('contain', 'When other members pledge a monetary or non-monetary contribution, they will appear here.')
     cy.getByDT('givingParagraph').should('contain', 'No one needs monetary contributions at the moment. You can still add non-monetary contributions if you would like.')
   })
 
-  it('It should add income detail modal', () => {
+  it('After swithing to need income, it should ask you how much you need', () => {
     cy.getByDT('openIncomeDetailModal').click()
     cy.getByDT('needsIncomeRadio').click()
     cy.getByDT('introIncomeOrPledge').should('contain', 'What\'s your monthly income?')
+  })
+
+  it('It should not let you ask for money if you have more than the basic income', () => {
     cy.getByDT('inputIncomeOrPledge').type(500)
     cy.getByDT('badIncome').should('contain', 'It seems your income is not lower than the group\'s mincome.')
+  })
+
+  it('After updating the income under the limit it should hide the error message', () => {
     cy.getByDT('inputIncomeOrPledge').clear().type(100)
     cy.getByDT('badIncome').should('not.be.visible')
+  })
+
+  it('After closing the modal it should dislay how much you need', () => {
     cy.getByDT('submitIncome').click()
     cy.getByDT('headerNeed').should('contain', 'You need $100')
+  })
+
+  it('The user should be inform that even if he can\'t pledge he can still contribute', () => {
```

Comment:
Same as before, I would merge the last 5 tests "User1 decides to switch income details to needing $100" with all the assertions (and inline comments) if needed.